include/nbl/asset/filters/kernels/WeightFunctions.h

```diff
@@ -0,0 +1,308 @@
+#ifndef _NBL_ASSET_MATH_FUNCTIONS_H_INCLUDED_
+#define _NBL_ASSET_MATH_FUNCTIONS_H_INCLUDED_
+
+#include <limits>
+#include <cstdint>
+
+namespace nbl::asset
+{
+
+struct SDiracFunction
+{
+	// We can use std::nextafter after C++23
+	constexpr static inline float min_support = -1e-6f;
+	constexpr static inline float max_support = +1e-6f;
+	constexpr static inline uint32_t k_smoothness = 0;
+
+	template<int32_t derivative=0>
+	static inline double weight(float x)
+	{
+		if (x != 0.f)
+			return 0.0;
+
+		if constexpr (derivative == 0)
+		{
+			return std::numeric_limits<double>::infinity();
+		}
+		else
+		{
+			static_assert(false);
+			return core::nan<double>();
+		}
+	}
+};
+
+// Standard Box function, symmetric, value in the support is 1, integral is 1, so support must be [-1/2,1/2)
+struct SBoxFunction
+{
+	constexpr static inline float min_support = -0.5f;
+	constexpr static inline float max_support = +0.5f;
+	constexpr static inline uint32_t k_smoothness = 0;
+
+	template <int32_t derivative = 0>
+	static inline double weight(float x)
+	{
+		if (x >= min_support && x < max_support)
+		{
+			if constexpr (derivative == 0)
+			{
+				return 1.0;
+			}
+			else
+			{
+				static_assert(false);
+				return core::nan<double>();
+			}
+		}
+		return 0.0;
+	}
+};
+
+// Standard Triangle function, symmetric, peak in the support is 1 and at origin, integral is 1, so support must be [-1,1)
+struct STriangleFunction
+{
+	constexpr static inline float min_support = -1.f;
+	constexpr static inline float max_support = +1.f;
+	constexpr static inline uint32_t k_smoothness = 0;
+
+	template <int32_t derivative = 0>
+	static inline double weight(float x)
+	{
+		if (x >= min_support && x < max_support)
+		{
+			if constexpr (derivative > 0)
+			{
+				// Derivative at 0 not defined.
+				static_assert(false);
+				return core::nan<double>();
+			}
+			else
+			{
+				return 1.0 - core::abs(x);
+			}
+		}
+		return 0.0;
+	}
+};
+
+// Truncated Gaussian function, with stddev = 1.0, if you want a different stddev then you need to scale it.
+struct SGaussianFunction
+{
+	constexpr static inline float min_support = -3.f;
+	constexpr static inline float max_support = +3.f;
```

Comment:
were the supports (truncation) hardcoded previously?

Reply:
Yep, look here in `master` https://github.com/Devsh-Graphics-Programming/Nabla/blob/258f3642c22e31bed6c0114418190ebf16db2d14/include/nbl/asset/filters/kernels/CGaussianImageFilterKernel.h#L19

Reply:
they weren't hardcoded, they were compile time tweakable

do compile time tweak via template OR runtime (if you can)

Reply:
Ah you're right.. they weren't hardcoded.

Reply:
solved it myself.